fedimint-api/src/db/mod.rs

```diff
@@ -56,16 +59,83 @@ pub trait DatabaseValue: Sized + SerializableDatabaseValue {
 }
 
 pub type PrefixIter<'a> = Box<dyn Iterator<Item = Result<(Vec<u8>, Vec<u8>)>> + Send + 'a>;
+type Notifications = DashMap<Box<[u8]>, Arc<Notify>>;
 
 #[async_trait]
 pub trait IDatabase: Debug + Send + Sync {
-    async fn begin_transaction(&self, decoders: ModuleDecoderRegistry) -> DatabaseTransaction;
+    async fn begin_transaction<'a>(&'a self) -> Box<dyn IDatabaseTransaction<'a> + Send + 'a>;
```

Comment:
Is this necessary? The whole point of introducing the DatabaseTransaction struct was to avoid dealing with Box<dyn ...> all over the code

Reply:
Yea, i don't like it. But I don't like passing `decoders` and `notifications` to the database impls.

Reply:
Ok I see now, this is fine with me.

Reply:
Wrap in a different newtype then.

Reply:
`DatabaseTransaction` is meant to be that newtype.